server/tests/src/main/scala/tapir/server/tests/ServerTests.scala

```diff
@@ -199,9 +199,12 @@ trait ServerTests[R[_], S, ROUTE] extends FunSuite with Matchers with BeforeAndA
       .post(uri"$baseUri/api/echo/multipart")
       .multipartBody(multipart("fruit", "pineapple"), multipart("amount", "120"))
       .send()
-      .map { r =>
-        r.unsafeBody should include regex "name=\"fruit\"\\s*pineapple apple"
-        r.unsafeBody should include regex "name=\"amount\"\\s*240"
+      .map { r: Response[String] =>
+        // TODO: Properly parse the multipart response
+        r.unsafeBody should include("name=\"fruit\"")
```

Comment:
didn't the existing regexes work?

Reply:
So in the case where there was no X-Auth header, the existing regexes didn't work for any reason I could figure out. I visually inspected the failing output and as far as I could tell, the regex should have matched, but it didn't.  I'm happy to investigate that further, but in my opinion it would be better (to avoid the x-auth issue below) to parse the multipart properly to check that everything is right rather than relying on regexes.

In the case of the X-Auth header, however, the test expected the X-Auth header to come after the Content-Disposition header, but the apache http library I used to build the multipart response put the X-Auth header before the Content-Disposition header.

Reply:
ah sure, we shouldn't rely on any specific ordering. Maybe you could paste the output that you are getting, so that we could debug this?

Reply:
Here's the output I got that seems like it should have matched:

```
"--DoC0A9kRqX-tJ_nIiPFXDlMcErLlA2o82Z6g4l0
Content-Disposition: form-data; name="fruit"
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

pineapple apple
--DoC0A9kRqX-tJ_nIiPFXDlMcErLlA2o82Z6g4l0
Content-Disposition: form-data; name="amount"
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

240
--DoC0A9kRqX-tJ_nIiPFXDlMcErLlA2o82Z6g4l0--
" did not include substring that matched regex name="fruit"\s*pineapple apple
```

And here's the X-Auth output, which unsurprisingly didn't match:

```
"--Cqszqo4NUKm7cnLPGB6qe_L7Ex7aE97qq
X-Auth: Some(12)
Content-Disposition: form-data; name="data"; filename="test4217776886502585135tapir"
Content-Type: text/plain
Content-Transfer-Encoding: binary

oiram hcaep
--Cqszqo4NUKm7cnLPGB6qe_L7Ex7aE97qq--
" did not include substring that matched regex name="data"\s*X-Auth: Some\(12\)\s*oiram hcaep
ScalaTestFailureLocation: tapir.server.tests.ServerTests at (ServerTests.scala:228)
org.scalatest.exceptions.TestFailedException: "--Cqszqo4NUKm7cnLPGB6qe_L7Ex7aE97qq
X-Auth: Some(12)
Content-Disposition: form-data; name="data"; filename="test4217776886502585135tapir"
Content-Type: text/plain
Content-Transfer-Encoding: binary

oiram hcaep
--Cqszqo4NUKm7cnLPGB6qe_L7Ex7aE97qq--
" did not include substring that matched regex name="data"\s*X-Auth: Some\(12\)\s*oiram hcaep
```

Reply:
I just now filed #116 -- I can try to do that before landing this, or do that after this, whatever you prefer.